src/plots/cartesian/axes.js

```diff
@@ -700,18 +700,44 @@ axes.calcTicks = function calcTicks(ax, opts) {
         };
 
         if(
-            !_has('%f') &&
-            !_has('%H') &&
-            !_has('%I') &&
-            !_has('%L') &&
-            !_has('%Q') &&
-            !_has('%S') &&
-            !_has('%s') &&
-            !_has('%X')
+            !_has('%f') && // microseconds as a decimal number [000000, 999999]
+            !_has('%L') && // milliseconds as a decimal number [000, 999]
+            !_has('%Q') && // milliseconds since UNIX epoch
+            !_has('%s') && // seconds since UNIX epoch
+            !_has('%S') && // second as a decimal number [00,61]
+            !_has('%M') && // minute as a decimal number [00,59]
+            !_has('%H') && // hour (24-hour clock) as a decimal number [00,23]
+            !_has('%I') && // hour (12-hour clock) as a decimal number [01,12]
+            !_has('%p') && // either AM or PM
+            !_has('%X')    // the locale’s time, such as %-I:%M:%S %p
         ) {
-            if(_has('%x') || _has('%d') || _has('%e') || _has('%j')) definedDelta = ONEDAY;
-            else if(_has('%B') || _has('%b') || _has('%m')) definedDelta = ONEAVGMONTH;
-            else if(_has('%Y') || _has('%y')) definedDelta = ONEAVGYEAR;
+            if(
+                _has('%d') || // zero-padded day of the month as a decimal number [01,31]
+                _has('%e') || // space-padded day of the month as a decimal number [ 1,31]
+                _has('%j') || // day of the year as a decimal number [001,366]
+                _has('%u') || // Monday-based (ISO 8601) weekday as a decimal number [1,7]
+                _has('%w') || // Sunday-based weekday as a decimal number [0,6]
+                _has('%x')    // the locale’s date, such as %-m/%-d/%Y
+            ) definedDelta = ONEDAY;
+            else if(
+                _has('%A') || // full weekday name
+                _has('%a') || // abbreviated weekday name
```

Comment:
Don't weekdays go in the `ONEDAY` bucket?

Reply:
Done in https://github.com/plotly/plotly.js/pull/5065/commits/9424347a37c95d660bd2e4b9677a33834c112583.